src/trackers/trackingParams.js

```diff
@@ -0,0 +1,374 @@
+function setCrawlParams (site) {
+    if (!crawlParams.paramString) {
+        crawlParams.params = site.searchParams
+    }
+
+    // set the param string 
+    if (!crawlParams.paramString && site && site.search) {
+        crawlParams.paramString = site.search.replace(/^\?/, '')
+        crawlParams.paramStringEncoded = encodeURIComponent(crawlParams.paramString)
+    }
+}
+
+// Cleanup the final data
+// - remove request and cookie sites/values
+// - calculate prevalence 
+// - get a list of top entities
+// - remove false positive param
+function cleanupFinalData () {
+    let topExampleSites = ''
+    if (config.topExampleSites) {
+        topExampleSites = new Set(JSON.parse(fs.readFileSync(config.topExampleSites, 'utf-8')))
+    }
+
+    for (const param of Object.keys(results)) {
+        if (results[param].cookies) {
+            delete results[param].cookies.cookieSites
+            delete results[param].cookies.cookieValues
+        }
+
+        if (results[param].requests3p) {
+            delete results[param].requests3p.requestSites
+            delete results[param].requests3p.requestValues
+        }
+
+        results[param].prevalence = +(results[param].exampleSites.length / totalSites).toFixed(3)
+        results[param].requests3p.prevalence = +(results[param].requests3p.prevalence / totalSites).toFixed(3)
+        results[param].cookies.prevalence = +(results[param].cookies.prevalence / totalSites).toFixed(3)
+
+        // calculate percents for entities
+        results[param].cookies.entities = getTopEntities(results[param].cookies.entities)
+        results[param].requests3p.entities = getTopEntities(results[param].requests3p.entities)
+
+        if (topExampleSites) {
+            const topParamSites = getTopExampleSites(results[param].exampleSites, topExampleSites)
+            results[param].exampleSites = _shuffleList(topParamSites, 10).map(x => x.replace(/\/$/, ''))
+        } else {
+            results[param].exampleSites = _shuffleList(results[param].exampleSites, 10).map(x => x.replace(/\/$/, ''))
+        }
+
+
+    }
+    
+    if (results[falsePositiveParams[0]]) {
+        delete results[falsePositiveParams[0]]
+    }
+}
+
+// Calculate entity percent and return sorted list of entities by percent 
+function getTopEntities(entities, numberOfEntities=10) {
+    const sortedEntityList = Object.entries(entities).sort((a,b) => {return b[1] - a[1]})
+
+    // total for all entities used to calculate percent
+    const entityTotal = sortedEntityList.reduce((total, e) => {
+        total += e[1]
+        return total
+    }, 0)
+
+    return sortedEntityList.slice(0, numberOfEntities - 1).reduce((obj, e) => {
+        obj[e[0]] = +(e[1] / entityTotal).toFixed(3)
+        return obj
+    }, {})
+}
+
+// look up entity data based on domain
+function _getEntity(url) {
+    return domainMap[url.domain]
```

Comment:
should we also check higher level domains? e.g. we start with "a.ble.com", should we also check "ble.com"  (like we often do in extensions)

Reply:
Don't need to. The domain map doesn't have subdomains in the keys. https://raw.githubusercontent.com/duckduckgo/tracker-radar/main/build-data/generated/domain_map.json

Reply:
Yeah true, we have a helper for this anyway. I'll use that instead.